### ISSUE TITLE
Help building

### ISSUE THREAD

**Issue by tocklime** [voidware/reckon#3]
I'm sorry if I'm doing something fundamentally wrong, here. I've installed the Casio fx-9860G SDK, cloned the project, loaded the project into the SDK, and hit build. I get the following output:

```
Executing Hitachi SH C/C++ Compiler/Assembler phase

set SHC_INC=C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\include
set PATH=C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin
set SHC_LIB=C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin
set SHC_TMP=C:\Users\greg\Documents\Code\playground\reckon\Debug
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 205
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk3226.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 222
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk3469.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 239
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk367D.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 256
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk391E.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 273
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk3B22.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 290
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk3E11.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 307
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk4064.tmp
	WARNING: The following dependant file(s) do not exist: "2d.h" "2di.h" "bcd.h" "bcdfloat.h" "bcdmath.h" "calc.h" "common.h" "complex.h" "dlist.h" "eeval.h" "mi.h" "plot.h" "l3d\primitiv.h" "region.h" "solve.h" "symbol.h" "types.h" "complex2.h" "finance.h". Line: 324
"C:\Program Files (x86)\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmk42B7.tmp
C:\Users\greg\Documents\Code\playground\reckon\cutils.h(50) : C2500 (E) Illegal token "int"
C:\Users\greg\Documents\Code\playground\reckon\cutils.h(50) : C2500 (E) Illegal token "const"
C:\Users\greg\Documents\Code\playground\reckon\cutils.h(50) : C2500 (E) Illegal token "n"
C:\Users\greg\Documents\Code\playground\reckon\cutils.h(51) : C2500 (E) Illegal token "{"
C:\Users\greg\Documents\Code\playground\reckon\common.c(40) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\Code\playground\reckon\common.c(42) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\Code\playground\reckon\common.c(43) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\Code\playground\reckon\common.c(44) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\Code\playground\reckon\common.c(45) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\Code\playground\reckon\common.c(601) : C1016 (W) Argument mismatch

HMAKE MAKE UTILITY Ver. 1.1
Copyright (C) Hitachi Micro Systems Europe Ltd. 1998
Copyright (C) Hitachi Ltd. 1998 


	ERROR: Process failed with return code: 1
Build was not successful.
```

Is there something obvious I'm missing?

**Comment by tocklime** [comment]
The files 2d.h, 2di.h, bcd.h, etc. all do actually exist in my clone as they do in the repository.

**Comment by voidware** [comment]
Hi,

Is this something to with with include paths. If the files are there, 
then perhaps that is the problem. Also, it's highly possible the SDK has 
changed.

-- hugh.

**Comment by tocklime** [comment]
Hmm. I can make the warnings go away by running 
```
& 'C:\users\greg\CASIO\fx-9860G SDK\OS\SH\BIN\HMAKE.EXE' .\Debug\Addin.mak
```
in the source directory (rather than from the `Debug` directory, which is what the auto-generated batch file does).

But the following errors still remain:
```
PS C:\Users\greg\Documents\code\reckon> & 'C:\users\greg\CASIO\fx-9860G SDK\OS\SH\BIN\HMAKE.EXE' .\Debug\Addin.mak

HMAKE MAKE UTILITY Ver. 1.1
Copyright (C) Hitachi Micro Systems Europe Ltd. 1998
Copyright (C) Hitachi Ltd. 1998

%3#C$z`&'0?

Executing Hitachi SH C/C++ Compiler/Assembler phase

set SHC_INC=C:\Users\greg\CASIO\fx-9860G SDK\OS\SH\include
set PATH=C:\Users\greg\CASIO\fx-9860G SDK\OS\SH\bin
set SHC_LIB=C:\Users\greg\CASIO\fx-9860G SDK\OS\SH\bin
set SHC_TMP=C:\Users\greg\Documents\code\reckon\Debug
"C:\Users\greg\CASIO\fx-9860G SDK\OS\SH\bin\shc.exe" -subcommand=C:\Users\greg\AppData\Local\Temp\hmkA95F.tmp
C:\Users\greg\Documents\code\reckon\cutils.h(50) : C2500 (E) Illegal token "int"
C:\Users\greg\Documents\code\reckon\cutils.h(50) : C2500 (E) Illegal token "const"
C:\Users\greg\Documents\code\reckon\cutils.h(50) : C2500 (E) Illegal token "n"
C:\Users\greg\Documents\code\reckon\cutils.h(51) : C2500 (E) Illegal token "{"
C:\Users\greg\Documents\code\reckon\common.c(40) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\code\reckon\common.c(42) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\code\reckon\common.c(43) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\code\reckon\common.c(44) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\code\reckon\common.c(45) : C1016 (W) Argument mismatch
C:\Users\greg\Documents\code\reckon\common.c(601) : C1016 (W) Argument mismatch

        ERROR: Process failed with return code: 1
```

Also I uninstalled the SDK from `C:\Program Files (x86)` and put it under my user directory, in case it's a strange new windows permission error. Didn't seem to make a difference though. Also setting various settings for compatability mode for `shc.exe` and `hmake.exe` doesn't seem to make a difference. 

I wonder if the spurious ``%3#C$z`&'0?`` in the output is relevant.

**Comment by voidware** [comment]
Sorry for the delay on this.

Do you know what version of the build tools you're using, i'll check 
against mine. Also, is there a newer set i can download. Perhaps it 
needs updating.

thanks,
-- hugh.

**Comment by watchmakerful** [comment]
I have the same set of errors. There is only one version of Casio SDK available. What's wrong with cutils.h?

**Comment by tocklime** [comment]
I never did get it to build. I had thought about trying it in a windows xp virtual machine, but didn't get round to it.

**Comment by voidware** [comment]
Hi,

I just reinstalled the 9860SDK. Sure enough, it doesn't build. So i 
think the version here in the repo is a newer version than the actual 
release. I made fixes so it compiles again. It runs on the emulator, at 
least.

Regarding the button mappings; this version is set for the "slim" 
edition, not the portrait model. I think that has different key mappings 
and i never got to find out what they were.

-- hugh.
